large-message-core/src/main/java/com/bakdata/kafka/AbstractLargeMessageConfig.java

```diff
@@ -260,13 +267,25 @@ private boolean enableAmazonS3PathStyleAccess() {
         final String roleExternalId = this.getString(S3_ROLE_EXTERNAL_ID_CONFIG);
         final String roleArn = this.getString(S3_ROLE_ARN_CONFIG);
         final String roleSessionName = this.getString(S3_ROLE_SESSION_NAME_CONFIG);
+        final String jwtPath = this.getString(S3_JWT_PATH_CONFIG);
 
-        if (!isEmpty(roleExternalId) && !isEmpty(roleArn) && !isEmpty(roleSessionName)) {
-            final AWSCredentialsProvider provider =
-                    new STSAssumeRoleSessionCredentialsProvider.Builder(roleArn, roleSessionName)
-                            .withStsClient(AWSSecurityTokenServiceClientBuilder.defaultClient())
-                            .withExternalId(roleExternalId)
-                            .build();
+        if (!isEmpty(roleArn) && !isEmpty(roleSessionName)) {
+            AWSCredentialsProvider provider = null;
```

Comment:
This will not work because the optional constructor does not accept null. You can just immediately return inside the ifs as it was before